HPL2/core/CMakeLists.txt

```diff
@@ -4,24 +4,27 @@ project(HPL2)
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 include(BoilerPlate)
 
-# Set up dependencies dirs (CACHE = global scope)
+# Set up dependency directories (CACHE = global scope)
 set(DEPENDENCIES_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/../../dependencies/precompiled CACHE PATH "Dir which holds precompiled dependencies - libs and headers")
 if(MINGW)
     set(DEPENDENCIES_PLATFORM_LIBS "${DEPENDENCIES_ROOT}/lib/mingw32;${DEPENDENCIES_ROOT}/lib/win32" CACHE PATH "Dir which holds prebuilt libraries for target system")
 else()
     set(DEPENDENCIES_PLATFORM_LIBS ${DEPENDENCIES_ROOT}/lib/${PLATFORM_PREFIX} CACHE PATH "Dir which holds prebuilt libraries for target system")
 endif()
 
+# Specify the static linking options for GCC and C++ runtimes
+if(MINGW)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -static-libgcc -static-libstdc++")
+endif()
+
 include(UtilityFunctions)
 
 add_subdirectory(../../dependencies/OALWrapper build_oalwrapper)
 
-# We use SDL2 by default in CMake now, but in the code 1.2 is also supported and we need to instruct it to use 2
-add_definitions(-DUSE_SDL2)
-add_definitions(-DUSE_OALWRAPPER)
+# We use SDL2 by default in CMake now, but SDL 1.2 is also supported in the code. We need to instruct it to use SDL2.
+add_definitions(-DUSE_SDL2 -DUSE_OALWRAPPER)
```

Comment:
Might as well change this to `add_compile_definitions(USE_SDL2 USE_OALWRAPPER)` because `add_definitions` is kinda deprecated

Reply:
Done.